plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,485 @@
+								{
+									detail: {
+										requestDurations,
+									},
+								}
+							)
+						);
+					} catch ( e ) {
+						console.error( e );
+					}
+					requestDurations = [];
+					if ( data.error ) {
+						switch ( data.error ) {
+							case 'expired_jwt_token':
+							case 'malformed_jwt_token':
+							case 'invalid_jwt_token':
+							case 'invalid_issuer':
+							case 'invalid_service':
+							case 'missing_jwt_token':
+								permanentlyDisabledServices.push( key );
+								return null;
+							case 'rate_limit_exceeded':
+								permanentlyDisabledServices.push( key );
+								setTimeout( () => {
+									permanentlyDisabledServices.splice(
+										permanentlyDisabledServices.indexOf(
+											key
+										),
+										1
+									);
+								}, response.headers.get( 'RateLimit-Retry-After' ) * 1000 );
+								return null;
+							case 'missing_address_id':
+								console.error(
+									'Automattic Address Suggestion: Missing address ID'
+								);
+								return null;
+							case 'no_place':
+								console.error(
+									'Automattic Address Suggestion: No place found'
+								);
+								return null;
+							case 'missing_session_id':
+								sessionId =
+									crypto && crypto.randomUUID
+										? crypto.randomUUID()
+										: Math.random()
+												.toString( 36 )
+												.substring( 2 );
+								return [];
+							case 'woo_addression_suggestion_internal_error':
+							case 'woo_addression_suggestion_server_error':
+								serviceErrorRetries++;
+								if (
+									serviceErrorRetries >=
+									MAX_SERVICE_ERROR_RETRIES
+								) {
+									permanentlyDisabledServices.push( key );
+									console.error(
+										'Automattic Address Suggestion has been disabled due to internal service error'
+									);
+								}
+								return null;
+							default:
+								console.error(
+									'Automattic Address Suggestion: Unknown error'
+								);
+								return null;
+						}
+					}
+
+					return data;
+				},
+			} );
+		}
+	);
+} )();
+
+function createBeacon( section, { name, value, type } ) {
+	const event = name.replace( '-', '_' );
+
+	// A counting event defaults to incrementing by one.
+	if ( type === 'counting' ) {
+		value = value === undefined ? 1 : value;
+	}
+	value = Array.isArray( value ) ? value : [ value ];
+	return value.map(
+		( v ) =>
+			`a8c.${ section }.${ event }:${ v }|${
+				type === 'timing' ? 'ms' : 'c'
+			}`
+	);
+}
+
+function createStatsdURL( sectionName, events ) {
+	if ( ! Array.isArray( events ) ) {
+		events = [ events ]; // Only a single event was passed to process.
+	}
+
+	const sanitizedSection = sectionName.replace( /[.:-]/g, '_' );
+	const json = JSON.stringify( {
+		beacons: events
+			.map( ( event ) => createBeacon( sanitizedSection, event ) )
+			.flat(),
+	} );
+
+	const encodedJson = encodeURIComponent( json );
+
+	return `https://pixel.wp.com/boom.gif?json=${ encodedJson }`;
+}
```

Comment:
This code is copied and modified from a calypso code, here we use Automattic's statsd library and we send several beacons, each with the time we had.